gluten-core/src/main/scala/io/glutenproject/extension/columnar/PullOutPreProject.scala

```diff
@@ -83,38 +79,53 @@ case class PullOutPreProject(session: SparkSession)
     }
   }
 
-  override def apply(plan: SparkPlan): SparkPlan = PhysicalPlanSelector.maybe(session, plan) {
-    plan.transform {
-      case sort: SortExec if needsPreProject(sort) =>
-        val expressionMap = new mutable.HashMap[Expression, NamedExpression]()
-        val newSortOrder = getNewSortOrder(sort.sortOrder, expressionMap)
+  override def apply(plan: SparkPlan): SparkPlan = {
+    val transformedPlan = plan.transform(applyLocally)
+    AddTransformHintRule().apply(transformedPlan)
```

Comment:
Can we move `AddTransformHintRule` out ? it seems we just apply `AddTransformHintRule` one more time after pull out pre project.

Reply:
Do you mean inserting `AddTransformHintRules` again in `preOverrides` after `PullOutPreProject`?

Reply:
yeah, just pull it out to make `preOverrides` clear

Reply:
OK. It is better.